functions.opengraph.php

```diff
@@ -293,17 +293,29 @@ function jetpack_og_get_image( $width = 200, $height = 200, $max_images = 4 ) {
 	if ( empty( $image ) && function_exists( 'blavatar_domain' ) ) {
 		$blavatar_domain = blavatar_domain( site_url() );
 		if ( blavatar_exists( $blavatar_domain ) ) {
-			$image_url = blavatar_url( $blavatar_domain, 'img', $width, false, true );
-
 			$img_width  = '';
 			$img_height = '';
-			$cached_image_id = get_transient( 'jp_' . $image_url );
-			if ( false !== $cached_image_id ) {
-				$image_id = $cached_image_id;
-			} else {
+
+			$image_url = blavatar_url( $blavatar_domain, 'img', $width, false, true );
+
+			// Build a hash of the Image URL. We'll use it later when building the transient.
+			if ( $image_url ) {
+				/**
+				 * Transient names are 45 chars max.
+				 * Let's generate a hash that's never more than 40 chars long.
+				 */
+				$image_hash = sha1( $image_url );
+			}
+
+			// Look for data in our transient. If nothing, let's get an attachment ID.
+			$cached_image_id = get_transient( 'jp_' . $image_hash );
+			if ( ! is_int( $cached_image_id ) ) {
```

Comment:
`attachment_url_to_postid()` casts its result as an integer so `get_transient()` should return an integer as well. If it doesn't, it means the value was modified and we probably shouldn't use it.

Since we use that transient in `wp_get_attachment_image_src()` later, and since `wp_get_attachment_image_src()` expects an integer, I think it's best to check for `is_int()` before to use the value.

Or do you think I should use `is_numeric()` and allow for third-parties to overwrite the transients saved by Jetpack and save strings instead?